packages/popover/src/vaadin-popover.js

```diff
@@ -127,17 +155,45 @@ class Popover extends PopoverPositionMixin(
    */
   _removeTargetListeners(target) {
     target.removeEventListener('click', this.__onTargetClick);
+    target.removeEventListener('keydown', this.__onTargetKeydown);
   }
 
   /** @private */
   __onTargetClick() {
     this._opened = !this._opened;
   }
 
+  /** @private */
+  __onTargetKeydown(event) {
+    if (event.key === 'Escape' && !this.noCloseOnEsc) {
+      this._opened = false;
+    }
+  }
+
```

Comment:
I wonder if this code is unnecessary. The tests pass and `noCloseOnEsc` seems to work even with the method content commented out.

Reply:
I think the tests currently pass because the overlay is modal by default. 
This makes me think #7412 should be merged first.